packages/amplify_datastore/example/integration_test/separate_integration_tests/basic_model_operation_test.dart

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:integration_test/integration_test.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:amplify_flutter/amplify_flutter.dart';
+
+import '../utils/setup_utils.dart';
+import '../utils/wait_for_expected_event_from_hub.dart';
+import 'models/basic_model_operation/ModelProvider.dart';
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  final enableCloudSync = shouldEnableCloudSync();
+  group(
+      'Basic model operation${enableCloudSync ? ' with API sync 🌩 enabled' : ''} -',
+      () {
+    setUpAll(() async {
+      await configureDataStore(
+          enableCloudSync: enableCloudSync,
+          modelProvider: ModelProvider.instance);
+    });
+
+    testWidgets(
+        'should save a new model ${enableCloudSync ? 'and sync to cloud' : ''}',
+        (WidgetTester tester) async {
+      Blog testBlog = Blog(name: 'test blog');
+
+      if (enableCloudSync) {
+        // set an async getter to retrieve a desired hub event with a speicfic
+        // event matcher
+        var eventGetter = getExpectedSubscriptionDataProcessedEvent(
+          eventMatcher: (event) =>
+              (event.element.model as Blog).id == testBlog.id &&
+              // newly saved model at this step should have version: 1
+              event.element.version == 1,
+        );
+
+        // save model locally and to sync to cloud
+        await Amplify.DataStore.save(testBlog);
+
+        // wait for the desired event to arrive
+        var event = await eventGetter;
+        expect(event.element.deleted, isFalse);
+      } else {
+        await Amplify.DataStore.save(testBlog);
+      }
+
+      var blogs = await Amplify.DataStore.query(Blog.classType);
+      expect(blogs.length, 1);
+      expect(blogs.contains(testBlog), isTrue);
+    });
+
+    testWidgets(
+      'should update existing model ${enableCloudSync ? 'and sync to cloud' : ''}',
+      (WidgetTester tester) async {
+        // get previously saved model
+        var testBlog = (await Amplify.DataStore.query(Blog.classType))[0];
```

Comment:
question: Could there be some kind of race condition/error getting just the first blog from DB/API if some other client running the test? For API integ tests, uses a utility to create instances for each test. Maybe that's too much here, just asking. Same question for other tests.

While race condition might seem unlikely, always consider the CI will run android/ios in parallel.